### PR TITLE
Add tab completion for action commands

### PR DESCRIPTION
**Pull request by adfoster-r7**
Adds support when tab completing options for action commands, specifically for the following scenarios:

```
use auxiliary/admin/http/gitstack_rest
run rhos<tab>
list rhos<tab>
```

Both will result in the tab completion `rhosts=`

## Verification

```
use auxiliary/admin/http/gitstack_rest
run rhos<tab>
list rhos<tab>
```

Ensure `rhosts=` is tab completed

### CODE REVIEW

File: lib/msf/ui/console/module_action_commands.rb [hunk 1]
```diff
@@ -29,24 +29,42 @@ def commands
...
       return do_action(action, *args)
     end
 
     super
   end
 
+  #
+  # Note: A change to this method will most likely require a corresponding change to method_missing
+  #
+  def respond_to_missing?(meth, _include_private = true)
```
Comment (adfoster-r7) [inline lib/msf/ui/console/module_action_commands.rb:55]: Context: https://thoughtbot.com/blog/always-define-respond-to-missing-when-overriding

And the corresponding calling logic which assumed `respond_to?` was implemented correctly:
https://github.com/rapid7/metasploit-framework/blob/c4639fa626cf04e24d7a2cfd5a9bcc66fdca04d2/lib/rex/ui/text/dispatcher_shell.rb#L467-L471

**Comment by gwillcox-r7** [comment]
Changes work as expected:

```
msf6 payload(windows/x64/meterpreter/reverse_tcp) > back
msf6 > use use auxiliary/admin/http/gitstack_rest

Matching Modules
================

   #  Name                                Disclosure Date  Rank    Check  Description
   -  ----                                ---------------  ----    -----  -----------
   0  auxiliary/admin/http/gitstack_rest  2018-01-15       normal  No     GitStack Unauthenticated REST API Requests


Interact with a module by name or index. For example info 0, use 0 or use auxiliary/admin/http/gitstack_rest

[*] Using auxiliary/admin/http/gitstack_rest
msf6 auxiliary(admin/http/gitstack_rest) > run r
run rhosts=  run rport=   
msf6 auxiliary(admin/http/gitstack_rest) > run rhosts=

[-] Msf::OptionValidateError The following options failed to validate: RHOSTS
msf6 auxiliary(admin/http/gitstack_rest) > list rhosts=

[-] Msf::OptionValidateError The following options failed to validate: RHOSTS
msf6 auxiliary(admin/http/gitstack_rest) > list r
list rhosts=  list rport=   
msf6 auxiliary(admin/http/gitstack_rest) > list rhosts=
```

**Comment by gwillcox-r7** [comment]
For reference before this PR here is what I get:

```
msf6 payload(windows/x64/meterpreter/reverse_tcp) > use auxiliary/admin/http/gitstack_rest
msf6 auxiliary(admin/http/gitstack_rest) > list r

[-] Msf::OptionValidateError The following options failed to validate: RHOSTS
msf6 auxiliary(admin/http/gitstack_rest) > run r
run rhosts=  run rport=   
msf6 auxiliary(admin/http/gitstack_rest) > run r
run rhosts=  run rport=   
msf6 auxiliary(admin/http/gitstack_rest) > list
list        list_repos  listm       
msf6 auxiliary(admin/http/gitstack_rest) > list r
```

**Review by gwillcox-r7 — APPROVED** [review]

**Comment by gwillcox-r7** [comment]
# Release Notes
The `lib/msf/ui/console/module_action_commands.rb` library has been enhanced to allow `msfconsole` to tab complete action commands, giving users greater flexibility in what they can autocomplete on the command line.

**Pull request merged**